File: lib_kotlin/src/main/java/lib/kalu/adapter/listener/OnDragChangeListener.java
package lib.kalu.adapter.listener;

import android.support.v7.widget.RecyclerView;

/**
 * description: 拖拽监听
 * created by kalu on 2017/5/26 15:14
 */
public interface OnDragChangeListener {

    void onDragStart(RecyclerView.ViewHolder viewHolder, int position);

    void onDragMove(RecyclerView.ViewHolder source, RecyclerView.ViewHolder target, int fromPosition, int toPosition);

    void onDragEnd(RecyclerView.ViewHolder viewHolder, int position);
}

File: lib_kotlin/src/main/java/lib/kalu/adapter/listener/OnDragChangeListener.kt
package lib.kalu.adapter.listener

import android.support.v7.widget.RecyclerView

/**
 * description: 拖拽监听
 * created by kalu on 2017/5/26 15:14
 */
interface OnDragChangeListener {

    fun onDragStart(viewHolder: RecyclerView.ViewHolder, position: Int)

    fun onDragMove(source: RecyclerView.ViewHolder, target: RecyclerView.ViewHolder, fromPosition: Int, toPosition: Int)

    fun onDragEnd(viewHolder: RecyclerView.ViewHolder, position: Int)
}